### DISCUSSION TITLE
Authors metadata in docs (possibly by git commiter?)

### DISCUSSION THREAD

**Discussion by kddige** [fuma-nama/fumadocs#979] [Q&A] [answered]
Hi! First of all, thanks for creating such a super sleek and DX friendly docs framework -, its been a pleasure to use!

I would really like to have authors signed at the bottom of each documentation page, so readers quickly can go ask X author about "why this", or "how that". -, is there:

A: a way to set author in the mdx metadata?
B: possibly (and preferred) a way to extract the files git information, and append the commiter(s) handle to the docs file on build? 🙊

(again) fkin thanks for keeping sanity in our codebase 💯

**Comment by Gitstar-OC** [comment]
Could you please elaborate ? 
- Means do you have a documentation site which will be written by multiple authors ?
- What platform would you like to use to ask the author ?
- Is it github, because you can add something like the below image in the layout.tsx file and with a link to change the file.  
![{852091F2-E70C-4B64-BACD-38DD3E2B29E8}](https://github.com/user-attachments/assets/0c6634f7-f2bc-4a72-b275-f5c78f2801f1)

**Reply by kddige** [reply]
Yo @Gitstar-OC, thanks for taking time to join in!

Documentation site will be written by our internal project maintainers, and occasionally other developers, that see it fit for their current task.

In the meantime, between i asked this question and now, i have implemented a function that uses the git local cli to extract commit author metadata, and uses that to get the users handle + name inside the organisation (using githubs api). It then appends the author information to the bottom of the page as normal html inside `app/docs/page.tsx` unde the `mdx` block.

Still interested in having authors defined manually in the metadata tho 😅

**Comment by Gitstar-OC** [comment]
Oh means do you want a feature like #878, to add the author name like the label I have added ? or a link to ask by whom it was written.

**Reply by kddige** [reply]
yes, but no jsx tho, just like:

```md
---
title: Some Docs Page
authors: John Doe, Jane Doe, Jospeh Doe
---
```

And then on the docs page it gets added to the footer as:


**Authors**
- John Doe
- Jane Doe
- Joseph Doe

**Reply by Gitstar-OC** [reply]
Yes, why you will need to do is you will need to modify the page.tsx file in the docs and import the custom tag and then show it, I was also trying to create it but forget later. I am busy with some other work right now. 

Please try to do it if you can, Notify me if you can do it or I will assist you after a week.

**Comment by fuma-nama** [comment, marked as answer]
Fumadocs MDX is a content source which essentially controls the properties exposed by frontmatter.

See https://fumadocs.vercel.app/docs/mdx/collections#schema-1 to customise the frontmatter schema, after that you can access them in your app and render it.

If you simply want to fetch them from version control APIs, maybe you can read the GitHub API docs.
